test/set.js

```diff
@@ -196,6 +196,12 @@ describe('Sets', () => {
     expect(array).toEqual([1, 2, 3])
   })
 
+  test('reduce', () => {
+    const set = Set.of([1, 2, 3, 4, 5])
+
+    expect(set.reduce((sum, value) => sum + value, 0)).toBe(15)
```

Comment:
Can you please add another assertion with an empty set?
Can you please add another assertion with a missing initial value?

Reply:
I was working on this, but jest doesn't want to work while I guess it should work, I'll commit the code. Can you check what I am doing wrong?

Reply:
Wait I think I got it

Reply:
I got some of it fixed, but it would be helpful if you could check what the error could be and what I did wrong in the tests.